backend/drm/drm.c

```diff
@@ -451,9 +451,24 @@ static void realloc_planes(struct wlr_drm_backend *drm, const uint32_t *crtc_in,
 			continue;
 		}
 
-		uint32_t possible[drm->num_type_planes[type] + 1];
-		uint32_t crtc[drm->num_crtcs + 1];
-		uint32_t crtc_res[drm->num_crtcs + 1];
+		uint32_t *possible = malloc(sizeof(uint32_t) * drm->num_type_planes[type]);
+		uint32_t *crtc = malloc(sizeof(uint32_t) * drm->num_crtcs);
+		uint32_t *crtc_res = malloc(sizeof(uint32_t) * drm->num_crtcs);
```

Comment:
I had something similar to your first example a few commits back, but clang wasn't happy with it. I think because some of the variables could've been freed before they were cleared in the `goto` block. Anyway - I much prefer your suggestion and clang doesn't complain about it, so I've made the change.